packages/eightballer/connections/ccxt_wrapper/connection.py

```diff
@@ -147,54 +146,38 @@ async def send(self, envelope: Envelope) -> None:
         self.executing_tasks.append(task)
         self.task_to_request[task] = envelope
 
-    async def receive(self, *args: Any, **kwargs: Any) -> Envelope | None:
-        """Receive an envelope."""
-        envelope = cast(
-            Envelope,
-            await self.queue.get(),
-        )
-        del args, kwargs
-        return envelope
+    async def receive(self, *args: Any, **kwargs: Any) -> Envelope:  # noqa: ARG002
+        """Receive a message."""
+        return cast(Envelope, await self.queue.get())
 
-    async def _execute(self, envelope=None) -> Message | None:
-        """Execute the task."""
-        dialogue: Dialogue
+    async def _execute(self, envelope: Envelope) -> Message:
         try:
-            dialogue = self.protocol_interface.validate_envelope(envelope)
+            self.protocol_interface.validate_envelope(envelope)
             return await self.protocol_interface.handle_envelope(envelope)
         except Exception as error:  # pylint: disable=broad-except
             self.logger.exception(f"Couldn't execute task, e={error} traceback={traceback.print_exc()}")
-            return self.get_error_message(error, envelope.message, dialogue)
+            return self.get_error_message(error, envelope.message)
 
-    def _handle_req(self, envelope) -> Task:
+    def _handle_req(self, envelope: Envelope) -> Task:
         """Create a task."""
         return self.loop.create_task(self._execute(envelope))
 
     def _handle_done_task(self, task: Task) -> None:
         """Handle completed task."""
-        request = self.task_to_request.pop(task, None)
+        request = self.task_to_request.pop(task)
         self.executing_tasks.remove(task)
         response_message: Message | None = task.result()
         response_envelope = self.protocol_interface.build_envelope(request, response_message)
-        self.logger.info(f"Placing {response_message} in queue")
         if response_envelope is None:
             return
-        self.logger.info(f"Placing {response_message} in queue")
+        self.logger.debug(f"Placing {response_message} in queue")
         self.queue.put_nowait(response_envelope)
```

Comment:
We don't need `None` in our pop. If the task does NOT exist, then we will crash on the next line `self.executing_tasks.remove(task)` because the task will not be in the `list` (which should be a `set`)